tests/test_all_simple_paths.py

```diff
@@ -0,0 +1,347 @@
+        paths = retworkx.dag_all_simple_paths(dag, 0, 5)
+        expected = [
+            [0, 1, 2, 3, 4, 5],
+            [0, 1, 2, 4, 5],
+            [0, 1, 3, 2, 4, 5],
+            [0, 1, 3, 4, 5],
+            [0, 2, 3, 4, 5],
+            [0, 2, 4, 5],
+            [0, 3, 2, 4, 5],
+            [0, 3, 4, 5]]
+        for i in expected:
+            self.assertIn(i, paths)
+
+    def test_all_simple_paths_min_depth(self):
+        dag = retworkx.PyDAG()
+        for i in range(6):
+            dag.add_node(i)
+        edges = [
+            (0, 1),
+            (0, 2),
+            (0, 3),
+            (1, 2),
+            (1, 3),
+            (2, 3),
+            (2, 4),
+            (3, 2),
+            (3, 4),
+            (4, 2),
+            (4, 5),
+            (5, 2),
+            (5, 3)]
+        for edge in edges:
+            dag.add_edge(edge[0], edge[1], None)
+        paths = retworkx.dag_all_simple_paths(dag, 0, 5, min_depth=6)
+        expected = [
+            [0, 1, 2, 3, 4, 5],
+            [0, 1, 3, 2, 4, 5],
+        ]
+        for i in expected:
+            self.assertIn(i, paths)
+
+    def test_all_simple_paths_with_cutoff(self):
+        dag = retworkx.PyDAG()
+        for i in range(6):
+            dag.add_node(i)
+        edges = [
+            (0, 1),
+            (0, 2),
+            (0, 3),
+            (1, 2),
+            (1, 3),
+            (2, 3),
+            (2, 4),
+            (3, 2),
+            (3, 4),
+            (4, 2),
+            (4, 5),
+            (5, 2),
+            (5, 3)]
+        for edge in edges:
+            dag.add_edge(edge[0], edge[1], None)
+        paths = retworkx.dag_all_simple_paths(dag, 0, 5, cutoff=4)
+        expected = [
+            [0, 2, 4, 5],
+            [0, 3, 4, 5]]
+        self.assertEqual(len(expected), len(paths))
+        for i in expected:
+            self.assertIn(i, paths)
+
+    def test_all_simple_paths_with_min_depth_and_cutoff(self):
+        dag = retworkx.PyDAG()
+        for i in range(6):
+            dag.add_node(i)
+        edges = [
+            (0, 1),
+            (0, 2),
+            (0, 3),
+            (1, 2),
+            (1, 3),
+            (2, 3),
+            (2, 4),
+            (3, 2),
+            (3, 4),
+            (4, 2),
+            (4, 5),
+            (5, 2),
+            (5, 3)]
+        for edge in edges:
+            dag.add_edge(edge[0], edge[1], None)
+        paths = retworkx.dag_all_simple_paths(dag, 0, 5, min_depth=5, cutoff=5)
+        expected = [
+            [0, 3, 2, 4, 5],
+            [0, 2, 3, 4, 5],
+            [0, 1, 3, 4, 5],
+            [0, 1, 2, 4, 5]
+        ]
+        self.assertEqual(len(expected), len(paths))
+        for i in expected:
+            self.assertIn(i, paths)
+
+    def test_all_simple_path_no_path(self):
+        dag = retworkx.PyDAG()
+        dag.add_node(0)
+        dag.add_node(1)
+        self.assertEqual([], retworkx.dag_all_simple_paths(dag, 0, 5))
```

Comment:
```suggestion
        self.assertEqual([], retworkx.dag_all_simple_paths(dag, 0, 1))
```
? Should it raise an error if `to` or `from` is not in the graph?

Reply:
Oh, good catch. Yeah we probably should raise here, I guess the petgraph function doesn't check this and just says there are no paths because it can't ever find a non-existent node index. I'll add a check for this before calling petgraph and update the test (and add a new one for this case).

Reply:
Fixed in: https://github.com/Qiskit/retworkx/pull/85/commits/0cc66afae85e70b67096306e4900137a1476a166